### PR TITLE
chore: clean up host CLI

### PR DESCRIPTION
**Pull request by clabby**
## Overview

Cleans up the host CLI validation by using the native features of the `clap::arg` proc macro over a custom validation function. Also adjusts some of the flag docs to be a bit more helpful.

and, colors, because it's kind of cool 🎨

Screenshot 2024-09-18 at 3 19 30 AM

### CODE REVIEW

**Comment by clabby** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/anton-rs/kona/536?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#536** <a href="https://app.graphite.dev/github/pr/anton-rs/kona/536?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#535** <a href="https://app.graphite.dev/github/pr/anton-rs/kona/535?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @clabby and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: bin/host/src/cli/tracing_util.rs [hunk 1]
```diff
@@ -6,17 +6,15 @@ use tracing::Level;
...
 pub fn init_tracing_subscriber(verbosity_level: u8) -> Result<()> {
     let subscriber = tracing_subscriber::fmt()
         .with_max_level(match verbosity_level {
-            0 => Level::ERROR,
-            1 => Level::WARN,
-            2 => Level::INFO,
-            3 => Level::DEBUG,
+            0 => Level::INFO,
+            1 => Level::DEBUG,
             _ => Level::TRACE,
```
Comment (refcell) [inline bin/host/src/cli/tracing_util.rs:18]: Why? This will require showing both warnings and info by default now. Is this intentional?

**Review by refcell — APPROVED** [review]
LGTM - though Q on the change in default tracing levels.

File: bin/host/src/cli/tracing_util.rs (hunk 1 above)
Comment (refcell) [reply bin/host/src/cli/tracing_util.rs:18]: Ya totally fine by me. Downstream users can roll their own CLI if they want to configure tracing differently.

**Pull request merged**